sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsClientBuilder.java

```diff
@@ -106,6 +127,14 @@ private static HttpPipeline buildPipeline(TokenCredential tokenCredential, Strin
 
         policies.addAll(additionalPolicies);
 
+        // If client options has headers configured, add a policy for each
+        if (clientOptions != null) {
+            List<HttpHeader> httpHeaderList = new ArrayList<>();
```

Comment:
Does it matter in Java (for memory optimization) to initialize a list with a known size ahead of time, and could you get that from `clientOptions.getHeaders()`?

Reply:
Good idea. ArrayList does have a constructor that takes the expected size

Reply:
But the getHeaders() call is just an iterable, so there isn't a way to know the size until iterated on, so that would be a wash